sdk/src/main/java/io/opentelemetry/sdk/trace/TraceProtoUtils.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import com.google.common.collect.EvictingQueue;
+import com.google.protobuf.ByteString;
+import io.opentelemetry.proto.trace.v1.Span.Attributes;
+import io.opentelemetry.proto.trace.v1.Span.Links;
+import io.opentelemetry.proto.trace.v1.Span.SpanKind;
+import io.opentelemetry.proto.trace.v1.Span.TimedEvents;
+import io.opentelemetry.proto.trace.v1.TruncatableString;
+import io.opentelemetry.resource.Resource;
+import io.opentelemetry.sdk.internal.TimestampConverter;
+import io.opentelemetry.sdk.trace.RecordEventsSpanImpl.AttributesWithCapacity;
+import io.opentelemetry.trace.AttributeValue;
+import io.opentelemetry.trace.Link;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.Status;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.Tracestate;
+import java.nio.charset.Charset;
+import java.util.Map;
+
+// Utilities to convert to trace protos.
+final class TraceProtoUtils {
+
+  private TraceProtoUtils() {}
+
+  static ByteString toProtoTraceId(TraceId traceId) {
+    return ByteString.copyFrom(traceId.toLowerBase16(), Charset.defaultCharset());
+  }
+
+  static ByteString toProtoSpanId(SpanId spanId) {
+    return ByteString.copyFrom(spanId.toLowerBase16(), Charset.defaultCharset());
+  }
+
+  static io.opentelemetry.proto.trace.v1.Span.Tracestate toProtoTracestate(Tracestate tracestate) {
+    io.opentelemetry.proto.trace.v1.Span.Tracestate.Builder builder =
+        io.opentelemetry.proto.trace.v1.Span.Tracestate.newBuilder();
+    for (Tracestate.Entry entry : tracestate.getEntries()) {
+      builder.addEntries(
+          io.opentelemetry.proto.trace.v1.Span.Tracestate.Entry.newBuilder()
+              .setKey(entry.getKey())
+              .setValue(entry.getValue()));
+    }
+    return builder.build();
+  }
+
+  static io.opentelemetry.proto.resource.v1.Resource toProtoResource(Resource resource) {
+    io.opentelemetry.proto.resource.v1.Resource.Builder builder =
+        io.opentelemetry.proto.resource.v1.Resource.newBuilder();
+    for (Map.Entry<String, String> entry : resource.getLabels().entrySet()) {
+      builder.putLabels(entry.getKey(), entry.getValue());
+    }
+    return builder.build();
+  }
+
+  static SpanKind toProtoKind(Kind kind) {
+    switch (kind) {
+      case CLIENT:
+        return SpanKind.CLIENT;
+      case SERVER:
+        return SpanKind.SERVER;
+    }
+    return SpanKind.UNRECOGNIZED;
+  }
+
+  static Attributes toProtoAttributes(AttributesWithCapacity attributes) {
```

Comment:
static Attributes toProtoAttributes(Map<String, AttributeValue> attributes, int droppedAttributes); This way it can be unified with the next method.

Reply:
SG, will do in the next PR.